### DISCUSSION TITLE
Developing formulae or casks - CodeMaster-formulae

### DISCUSSION THREAD

**Discussion by N3v1** [Homebrew/discussions#4793] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
none
```

### Output of `brew doctor`

```shell
none
```

### Description of issue

Hello guys, I have a small question. How can I "contribute" my own formulae? I'm currently working on my own code editor called "CodeMaster" and want to release my own CLI package. Can someone help me?

In the end i want that the users can install the package via homebrew with the following command: 
```sh
brew install codemaster
```

**Comment by gromgit** [comment, marked as answer]
Start by reading https://docs.brew.sh/Adding-Software-to-Homebrew

**Comment by p-linnane** [comment]
We do not accept formulae submitted by their authors unless they are incredibly popular (which would mean they are most likely already included in homebrew-core). Additionally, we have notability requirements for new formulae (>30 forks, >30 watchers and >75 stars), so these would need to be fulfilled before you submit a new formula as well. In short, you most likely will need to host this in your own tap until it becomes popular enough to pass notability requirements and be submitted by one of your users.
